storybookjs/storybook
# NextJS: Verify Next runs on desired port & fail fast

## What I did

NextJS doesn't start on the  desired port if that port is already being used by another process. However our Storybook setup assumes that it is being run on the specified port and will fail if not. This check exits the process if another port is being used.

This is a strawman for discussion and shouldn't be merged as is. The code actually gets called twice with two different PIDs since `next-config` gets evaluated twice. Consequently there is a race condition.

#### Manual testing

1. Verify the dev server starts normally
2. Start a second dev server while the first one is running and verify that the process exits early

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
